File: ParcAttractionSpring/src/main/java/SopraAjc/ParcAttractionSpring/model/Boutique.java
package SopraAjc.ParcAttractionSpring.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Boutique {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@JsonView(JsonViews.Common.class)
	private Long id;
	@JsonView(JsonViews.Common.class)
	private String nom;
	
	@ManyToMany
	@JsonView(JsonViews.BoutiqueMarchandise.class)
	private List <Marchandise> enVente = new ArrayList();
	
	@Embedded
	@JsonView(JsonViews.Common.class)
	private Coordonnees coordonnees;
	
	public Boutique() {
		
	}

	public Boutique(String nom,List<Marchandise> enVente, Coordonnees coordonnees) {
		this.nom=nom;
		this.enVente = enVente;
		this.coordonnees = coordonnees;
	}



	public Boutique(List<Marchandise> enVente) {
		this.enVente = enVente;
	}

	public List<Marchandise> getEnVente() {
		return enVente;
	}

	
	public void setEnVente(List<Marchandise> enVente) {
		this.enVente = enVente;
	}

	public Long getId() {
		return id;
	}

	public Coordonnees getCoordonnees() {
		return coordonnees;
	}

	public void setCoordonnees(Coordonnees coordonnees) {
		this.coordonnees = coordonnees;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}
	
	
	
	
	
}

File: parcAngular/src/app/model/boutique.ts
import { Coordonnees } from './coordonnees';
import { Marchandise } from './marchandise';

export class Boutique {
  public constructor(
    private _id?: number,
    private _nom?: string,
    private _description?: string,
    private _coordonnees?: Coordonnees,
    private _enVente?: Marchandise[]
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }

  public get coordonnees(): Coordonnees | undefined {
    return this._coordonnees;
  }

  public set coordonnees(value: Coordonnees | undefined) {
    this._coordonnees = value;
  }

  public get nom(): string | undefined {
    return this._nom;
  }

  public set nom(value: string | undefined) {
    this._nom = value;
  }

  public get description(): string | undefined {
    return this._description;
  }

  public set description(value: string | undefined) {
    this._description = value;
  }

  public get enVente(): Marchandise[] | undefined {
    return this._enVente;
  }

  public set enVente(value: Marchandise[] | undefined) {
    this._enVente = value;
  }
}
